src/debugger/channel/MessageMarshaller.js

```diff
@@ -105,6 +126,29 @@ export class MessageMarshaller {
     }
   }
 
+  unmarshallScopesResponse(requestID: number, responseBody: string): DebuggerResponse {
+    try {
+      let scopes = JSON.parse(responseBody);
+      invariant(Array.isArray(scopes), "Scopes is not an array");
+      for (const scope of scopes) {
+        invariant(scope.hasOwnProperty("name"), "Scope is missing name");
+        invariant(scope.hasOwnProperty("variablesReference"), "Scope is missing variablesReference");
+        invariant(scope.hasOwnProperty("expensive"), "Scope is missing expensive");
```

Comment:
What is expensive actually used for?

Reply:
It's specified by the protocol to tell the UI if this collection of variables is big or slow to retrieve and the UI can choose to retrieve them differently.